### PR TITLE
Add helpful error messages when using enum, indexed types/interfaces and maps

### PR DESCRIPTION
**Pull request by fwouts**
At the moment, we don't support the following types:
```typescript
// Enums are not supported at all.
enum SomeEnum {
  A,
  B,
  C
}

// Interfaces and types where keys are not fixed.

// 1) Interface syntax.
interface DictionaryInterface {
  [key: string]: string;
}

// 2) Type alias syntax.
type DictionaryType = {
  [key: string]: string;
}

// ES6 Maps.
type MapType = Map<string, string>;
```

Unfortunately, someone who uses Spot for the first time may assume that these may work. Therefore we introduce explicit errors so that it fails in a helpful way.

We may separately consider adding support for these syntaxes in the future.

### CODE REVIEW

File: lib/src/parsers/utilities/parser-utility.ts [hunk 1]
```diff
@@ -462,12 +469,43 @@ export function getTargetDeclarationFromTypeReference(
     }
   }
   const targetDeclaration = declarations[0];
+  if (
+    TypeGuards.isInterfaceDeclaration(targetDeclaration) &&
+    targetDeclaration.getIndexSignatures().length > 0
```
Comment (Kastet) [inline lib/src/parsers/utilities/parser-utility.ts:475]: What is an index signature? And why checking for them here?

File: lib/src/parsers/utilities/parser-utility.ts [hunk 2]
```diff
@@ -462,12 +469,43 @@ export function getTargetDeclarationFromTypeReference(
...
   }
+  if (TypeGuards.isEnumDeclaration(targetDeclaration)) {
+    throw new Error(
+      `enums are not supported (offending type: ${targetDeclaration.getName()})`
+    );
+  }
+  if (TypeGuards.isEnumMember(targetDeclaration)) {
+    throw new Error(
+      `enums are not supported (offending type: ${targetDeclaration
+        .getParent()
```
Comment (Kastet) [inline lib/src/parsers/utilities/parser-utility.ts:509]: What's `parent` for and why only in enums?

**Review by lfportal — APPROVED** [review]

File: lib/src/parsers/utilities/parser-utility.ts (hunk 1 above)
Comment (fwouts) [reply lib/src/parsers/utilities/parser-utility.ts:475]: Added clarifying comments :)

File: lib/src/parsers/utilities/parser-utility.ts (hunk 2 above)
Comment (fwouts) [reply lib/src/parsers/utilities/parser-utility.ts:509]: The parent of an enum member is the enum type itself. For example when the type is `SomeEnum.A`, we want to show `"offending type: SomeEnum"`.

**Review by Kastet — APPROVED** [review]

**Pull request merged**